src/extremeweatherbench/evaluate.py

```diff
@@ -2,19 +2,183 @@
 
 import logging
 import fsspec
-from typing import Optional, Any
+from typing import Optional, Any, Literal, Union
 import pandas as pd
 import xarray as xr
 from extremeweatherbench import config, events, case, utils
 import dacite
-
+import dataclasses
+import itertools
 
 logger = logging.getLogger(__name__)
 logger.setLevel(logging.INFO)
 
 #: Default mapping for forecast dataset schema.
 DEFAULT_FORECAST_SCHEMA_CONFIG = config.ForecastSchemaConfig()
 
+POINT_OBS_STORAGE_OPTIONS = dict(token="anon")
+
+GRIDDED_OBS_STORAGE_OPTIONS = dict(token="anon")
+
+
+@dataclasses.dataclass
+class CaseEvaluationInput:
+    """
+    A dataclass for storing the inputs of an evaluation.
+
+    Attributes:
+        observation_type: The type of observation to evaluate (gridded or point).
+        observation: The observation dataarray to evaluate.
+        forecast: The forecast dataarray to evaluate.
+    """
+
+    observation_type: Literal["gridded", "point"]
+    observation: Optional[xr.DataArray] = None
+    forecast: Optional[xr.DataArray] = None
+
+    def compute(self):
+        """Load the evaluation inputs into memory."""
+        self.observation = self.observation.compute()
+        self.forecast = self.forecast.compute()
+
+
+class CaseEvaluationData:
+    """
+    This class is designed to be used in conjunction with the `case.IndividualCase` class to build
+    datasets for evaluation of individual Cases across a set of Events.
+
+    Attributes:
+        individual_case: The `case.IndividualCase` object to evaluate.
+        forecast: The forecast dataset to evaluate.
+        gridded_observation: The gridded observation dataset to evaluate.
+        point_observation: The point observation dataset to evaluate.
+    """
+
+    def __init__(
+        self,
+        individual_case: case.IndividualCase,
+        observation_type: Literal["gridded", "point"],
+        forecast: xr.Dataset,
+        observation: Optional[Union[xr.Dataset | pd.DataFrame]],
+    ):
+        self.observation_type = observation_type
+        self.observation = observation
+        self.forecast = forecast
+        self.individual_case = individual_case
```

Comment:
This could (and probably should) also be a `dataclasses.dataclass`. It's basically a wrapper to ingest some datasets and package them together. Then, you can use it as a convenient input to a few functions below (see following comments).